### ISSUE TITLE
Feature Request — iOS Timer Parameter

### ISSUE THREAD

**Issue by tsinis** [csdcorp/speech_to_text#92]
Hey @sowens-csd!

It's not an Issue but feature request, or enchantment, as you labeling it. As you know, for some reason, Android version have no continuous recognition —  it will stop listening after around 1sec speech pause. Thing is, to add Boolean parameter to .listen method, so iOS speech recognition will have same behavior as Android one. At this moment it have to be implemented with Timer, something similar to this (this code is not working on my app sadly ATM, i have it from earlier version of another package):
```
Timer _onTimer;
SpeechToText _speech;
...
    if (Platform.isIOS) {
      _onTimer?.cancel();
      _onTimer = Timer(
        const Duration(seconds: 1),
        () => _speech.stop(),
      );
    }
...
```
Which is kinda ugly, i can imagine it to be implemented in your package like:
```
  void _start() => _speech.listen(
      ...
      partialResults: false,
      iOSTimer: true,
);
...
```
It should be false by default, so it won't break existing apps. Anyways — big thanks for this package, it's core of my app and it would be impossible without speech_to_text!

**Comment by sowens-csd** [comment]
The listenFor parameter on the listen method is very close I think? Or do you want it to only timeout on a pause in speech of a specified duration?

**Comment by tsinis** [comment]
> 
> 
> The listenFor parameter on the listen method is very close I think? Or do you want it to only timeout on a pause in speech of a specified duration?

Hey Stephen! Thanks for reply. Well, listenFor parameter allows you to listen only for specific amount of time, for example ten seconds, but I am talking about something else. I've created a very short video for you to be more clear:
https://youtu.be/e0JGlt_W7Rs
As you can see, iOS will never stop listening until you tap on stop button on your iPhone, and it will continue listening until your battery goes off. Android otherwise will stop listening after speech pause longer then 1sec (in video I'm saying word "test" five times, but time interval between 4th and 5th word is longer then 1 second, so it will automatically stop listening and only show result for 4 words).
  I suppose we cannot repeat behavior of iOS on Android side, due to some restrictions of Google's OS. Therefore, it would be great to at least be able to repeat the behavior of Android on the iOS platform, so that users would have the same UX using same application on both platforms.

P.S. it seems if in my example code above I'll set the partialResults to **true**,
that behavior will be exactly the same on both iOS and Android with additional Timer (as in my example code). I will try to repeat it within 12 hours and let you know.

**Comment by sowens-csd** [comment]
Got it, thank for the explanation. Yes, this would be a good improvement on the listenFor parameter. I'll add it in a future version. Thanks for the request!

**Comment by tsinis** [comment]
> Got it, thank for the explanation. Yes, this would be a good improvement on the listenFor parameter. I'll add it in a future version. Thanks for the request!

Thanks Stephen!
So as I wrote before: if partialResults is set to true my code works as it have to. I've added this behavior to your example (in this package repository), just swap main.dart with this one:
https://gist.github.com/tsinis/f0b8bf66cb8c07552ae9124411ad6c3b
as you can see, I've added at the top:
[`import 'dart:io' show Platform;`](https://gist.github.com/tsinis/f0b8bf66cb8c07552ae9124411ad6c3b#file-main-dart-L1)
then set
[partialResults: to true](https://gist.github.com/tsinis/f0b8bf66cb8c07552ae9124411ad6c3b#file-main-dart-L212)
and added timer to resultListener:
  ```
Timer _onTimer;
  void resultListener(SpeechRecognitionResult result) {
        if (Platform.isIOS) {
      _onTimer?.cancel();
      _onTimer = Timer(
        const Duration(seconds: 1),
        () => speech.stop(),
      );
    }
     ...
```
https://gist.github.com/tsinis/f0b8bf66cb8c07552ae9124411ad6c3b#file-main-dart-L229-L237

Anyways it would be nice to have this behavior even with partialResults: set to false, but I'm not sure it's possible. You are free to close this topic if you wish to, or leave it as remainder, it's up to you Stephen! **Thank you once again!**

**Comment by sowens-csd** [comment]
This feature is now available in the repo, will be released in 2.3.0. See the `pauseFor` parameter to the `listen` method.
